tasks/build/download_node_builds.js

```diff
@@ -68,17 +68,75 @@ module.exports = function (grunt) {
     var bytes = parseInt(resp.headers['content-length'], 10) || 'unknown number of';
     var mb = ((bytes / 1024) / 1024).toFixed(2);
     grunt.log.ok(`downloading ${platform.name} - ${mb} mb`);
+
+    await platform.downloadPromise;
+    grunt.log.ok(`${platform.name} download complete`);
+    let file = platform.nodeUrl.substr(platform.nodeUrl.lastIndexOf('/') + 1);
+    let path = platform.nodeDir + '/' + file;
+    let expected = platform.win ? shaSums['win-x86/' + file] : shaSums[file];
+    checkHash.fromFile(path, expected, function (err, passed, actual) {
+      if (err) {
+        throw (new Error(`${file} hash check failed: ${err.message}`));
+      } else if (!passed) {
+        throw (new Error(`${file} hash is not valid: ${expected} :: ${actual}`));
+      } else {
+        if (!platform.win) {
+          createReadStream(path)
+            .pipe(createGunzip())
+            .on('error', (err) => {
+              grunt.log.ok('error extracting: ' + err.message);
+            })
+            .pipe(new Extract({path: platform.nodeDir, strip: 1}))
+            .on('error', (err) => {
+              grunt.log.ok('error extracting: ' + err.message);
+            });
+        }
+      }
+    });
   };
 
-  grunt.registerTask('_build:downloadNodeBuilds:start', function () {
+  let getShaSums = async () => {
+    let nodeVersion = grunt.config.get('nodeVersion');
+    let shaSumsUri = `https://nodejs.org/dist/v${nodeVersion}/SHASUMS256.txt`;
+
+    let resp = await fromNode(cb => {
+      let req = wreck.request('GET', shaSumsUri, null, function (err, resp) {
+        if (err) {
+          return cb(err);
+        }
+
+        if (resp.statusCode !== 200) {
+          return cb(new Error(`${shaSumsUri} failed with a ${resp.statusCode} response`));
+        }
+
+        cb(null, resp);
+      });
+    });
+
+    let respBody = await fromNode(cb => {
+      var content = '';
+      resp
+        .setEncoding('utf8')
+        .on('data', function (chunk) {
+          content += chunk;
+        })
+        .on('error', cb)
+        .on('end', () => {
+          return cb(null, content);
+        });
+    });
+
+    respBody.split('\n').forEach(line => {
+      let cells = line.split('  ');
+      shaSums[cells[1]] = cells[0];
+    });
+
+    grunt.log.ok(`sha sums download complete`);
+  };
+
+  grunt.registerTask('_build:downloadNodeBuilds', function () {
     map(platforms, start).nodeify(this.async());
+    getShaSums();
```

Comment:
Shouldn't we download the checksums first? This looks like a race condition waiting to happen. And I think ideally we'd pass them in as an argument rather than using a mutable variable in the closure.